tests/parser/tst_parser.cpp

```diff
@@ -101,6 +112,101 @@ private slots:
     void recursionLimit();
 };
 
+struct ParserWrapper
+{
+    void *realdata = nullptr;
+    uint8_t *data;
+    size_t len;
+    CborParser parser;
+    CborValue first;
+
+    ~ParserWrapper() { freeMemory(); }
+
+    CborError init(const QByteArray &ba, uint32_t flags = 0)
+    {
+        return init(ba.constData(), ba.size(), flags);
+    }
+    CborError init(const char *ptr, int n, uint32_t flags = 0)
+    {
+        freeMemory();
+        data = allocateMemory(n);
+        memcpy(data, ptr, len);
+        return cbor_parser_init(data, len, flags, &parser, &first);
+    }
+    uint8_t *begin() { return data; }
+    uint8_t *end()   { return data + len; }
+
+    uint8_t *allocateMemory(size_t);
+    void freeMemory();
+
+    static const size_t PageSize = 4096;
```

Comment:
Is it good to assume 4096 is always the size of a page for portable code? Shouldn't you get it from sysconf(_SC_PAGESIZE)?

Reply:
tinycbor is portable to a lot more systems than its tests are. I really don't expect the tests to be run on anything except x86 and maybe ARM.

Reply:
Not a big deal then.